packages/x-components/src/plugins/x-plugin.mixin.ts

```diff
@@ -60,11 +61,15 @@ export function getBusAPI(bus: XBus, component: PrivateExtendedVueComponent): XC
     emit: <Event extends XEvent>(
       event: Event,
       payload?: XEventPayload<Event>,
-      metadata: Omit<WireMetadata, 'moduleName'> = {}
+      metadata: Omit<WireMetadata, 'moduleName' | 'origin'> = {}
     ) => {
       const xComponent = component.xComponent;
-      const moduleName = xComponent ? getXComponentXModuleName(xComponent) : null;
-      bus.emit(event, payload as any, { moduleName, origin: component.$origin, ...metadata });
+      bus.emit(event, payload as any, {
+        moduleName: xComponent ? getXComponentXModuleName(xComponent) : null,
+        origin: createQueryOrigin(metadata.feature, component.$location as QueryLocation),
+        location: component.$location,
```

Comment:
I'm not sure about using the `location` for tracking result events. I think we are mixing again the feature with the location we place it. For me it is clearer to see `recommendations:predictive_layer`, `recommendations:no_results`, rather than `no_results`, and having to think what feature are we using in the no_results? Is it the recommendations? Or viceversa, seeing `recommendations` without an `origin` and thinking _"hmm, they haven't origin so they might have no query associated. I guess these are the recommendations that we paint in the `no_query` location"_